src/posix/main.c

```diff
@@ -56,19 +59,155 @@
+{
+    fprintf(aStream,
+            "Syntax:\n"
+            "    %s [Options] NodeId|Device|Command [DeviceConfig|CommandArgs]\n"
+            "Options:\n"
+            "    -I  --interface-name name   Thread network interface name.\n"
+            "    -n  --dry-run               Just verify if arguments is valid and radio spinel is compatible.\n"
+            "        --no-reset              Do not reset RCP on initialization\n"
+            "        --radio-version         Print radio firmware version\n"
+            "    -s  --time-speed factor     Time speed up factor.\n"
+            "    -v  --verbose               Also log to stderr.\n"
+            "    -h  --help                  Display this usage information.\n",
+            aProgramName);
+    exit(aExitCode);
+}
+
+static otInstance *InitInstance(int aArgCount, char *aArgVector[])
+{
+    otPlatformConfig config;
+    otInstance *     instance          = NULL;
+    bool             isDryRun          = false;
+    bool             printRadioVersion = false;
+    bool             isVerbose         = false;
+
+    memset(&config, 0, sizeof(config));
+
+    config.mSpeedUpFactor = 1;
+    config.mResetRadio    = true;
+
+    optind = 1;
+
+    while (true)
+    {
+        int index  = 0;
+        int option = getopt_long(aArgCount, aArgVector, "hI:ns:v", kOptions, &index);
+
+        if (option == -1)
+        {
+            break;
+        }
+
+        switch (option)
+        {
+        case 'h':
+            PrintUsage(aArgVector[0], stdout, OT_EXIT_SUCCESS);
+            break;
+        case 'I':
+            config.mInterfaceName = optarg;
+            break;
+        case 'n':
+            isDryRun = true;
+            break;
+        case 's':
+        {
+            char *endptr          = NULL;
+            config.mSpeedUpFactor = (uint32_t)strtol(optarg, &endptr, 0);
+
+            if (*endptr != '\0' || config.mSpeedUpFactor == 0)
+            {
+                fprintf(stderr, "Invalid value for TimerSpeedUpFactor: %s\n", optarg);
+                exit(OT_EXIT_INVALID_ARGUMENTS);
+            }
+            break;
+        }
+        case 'v':
+            isVerbose = true;
+            break;
+
+        case 0:
+            if (!strcmp(kOptions[index].name, "radio-version"))
+            {
+                printRadioVersion = true;
+            }
+            else if (!strcmp(kOptions[index].name, "no-reset"))
+            {
+                config.mResetRadio = false;
+            }
+            break;
+        case '?':
+            PrintUsage(aArgVector[0], stderr, OT_EXIT_INVALID_ARGUMENTS);
+            break;
+        default:
+            assert(false);
+            break;
+        }
+    }
+
+#if OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED
+    openlog(aArgVector[0], LOG_PID | (isVerbose ? LOG_PERROR : 0), LOG_DAEMON);
+    setlogmask(setlogmask(0) & LOG_UPTO(LOG_DEBUG));
+#endif
```

Comment:
Would it be better to put the code between line 165 and 168 into a function?

Reply:
any motivation?

Reply:
The platform contains the file `logging.c`, If we put logging related code to this file, it will make the logic of the code clearer.

Reply:
This PR tries to build a library from `src/posix/platform/*`, so let the caller decide how to initialize syslog.

Reply:
Great.